### ISSUE TITLE
Sorting works only when manually reload confings

### ISSUE THREAD

**Issue by Disketaa** [SolidBlock-cn/reasonable-sorting#8]
When you entering the world the tab sort sets to default one, but if you change something in config and save it, then sorting applies

```
{
  "enableSorting": true,
  "enableDefaultItemSortingRules": true,
  "customSortingRules": [
    "minecraft:black_stained_glass_pane seasons:greenhouse_glass",
    "minecraft:stonecutter yigd:grave",
    "minecraft:dropper redstonebits:checker redstonebits:placer redstonebits:breaker minecraft:lightning_rod",
    "minecraft:target minecraft:hopper omnihopper:omnihopper omnihopper:fluid_hopper omnihopper:fluid_omnihopper goldenhoppers:golden_hopper",
    "minecraft:comparator redstonebits:counter redstonebits:adder redstonebits:resistor redstonebits:inverter",
    "minecraft:redstone_lamp redstonebits:analog_redstone_lamp minecraft:noteblock redstonebits:redstone_display",
    "minecraft:redstone_block automated_crafting:auto_crafter",
    "minecraft:sticky_piston redstonebits:rotator",
    "minecraft:lever minecraft:tripwire_hook",
    "minecraft:warped_button redstonebits:copper_button redstonebits:exposed_copper_button redstonebits:weathered_copper_button redstonebits:oxidized_copper_button redstonebits:waxed_copper_button redstonebits:waxed_exposed_copper_button redstonebits:waxed_weathered_copper_button redstonebits:waxed_oxidized_copper_button",
    "minecraft:light_weighted_pressure_plate redstonebits:medium_weighted_pressure_plate redstonebits:exposed_medium_weighted_pressure_plate redstonebits:weathered_medium_weighted_pressure_plate redstonebits:oxidized_medium_weighted_pressure_plate redstonebits:waxed_medium_weighted_pressure_plate redstonebits:waxed_exposed_medium_weighted_pressure_plate redstonebits:waxed_weathered_medium_weighted_pressure_plate redstonebits:waxed_oxidized_medium_weighted_pressure_plate",
    "minecraft:daylight_detector seasons:season_detector",
    "minecraft:hopper_minecart goldenhoppers:golden_hopper_minecart",
    "minecraft:heart_of_the_sea origins:orb_of_origin",
    "minecraft:fishing_rod minecraft:compass seasons:season_calendar minecraft:clock",
    "minecraft:leather_horse_armor villager-hats:fletcher_hat villager-hats:shepherd_hat villager-hats:butcher_hat villager-hats:fisherman_hat villager-hats:farmer_hat villager-hats:armorer_hat villager-hats:librarian_hat",
    "minecraft:"
  ],
  "variantsFollowingBaseBlocks": "stairs slab",
  "shapesFollowingBaseBlocks": "*",
  "fenceGateFollowsFence": true,
  "enableGroupTransfer": true,
  "buttonsInDecorations": false,
  "fenceGatesInDecorations": true,
  "swordsInTools": false,
  "doorsInDecorations": true,
  "transferRules": [],
  "variantTransferRules": [],
  "regexTransferRules": [
    ".+?hat misc"
  ],
  "shapeTransferRules": [],
  "baseBlocksInBuildingBlocks": true
}
```

**Comment by ardissaps** [comment]
I can confirm this.

**Comment by ardissaps** [comment]
![image](https://user-images.githubusercontent.com/82106625/175842915-d335be21-80cf-4bd2-9ab4-6f61f671ae42.png)
everytime login the transfered item back to their tab. 
![image](https://user-images.githubusercontent.com/82106625/175843045-33250ea3-b63c-4dba-9190-c2099045f2ca.png)

**Comment by ardissaps** [comment]
should have moved to building block according to my setting 
![image](https://user-images.githubusercontent.com/82106625/175842998-a3a5de30-5140-444b-995b-19077b9ddda6.png)

**Comment by SolidBlock-cn** [comment]
The issue will be fixed in the next release.

**Comment by ardissaps** [comment]
Thank you, I hope the fix available for 1.18.2. :p
